doc/delivery-monitoring-protocol.md

```diff
@@ -0,0 +1,46 @@
+
+## PROBLEM
+  There are [`TRANSPORT`](#terms)s that can fall and in this case, does not provide information about [`FRAGMENT`](#terms)s were delivered. This creates a problem after returning the ability to transfer fragments: which [`FRAGMENT`](#terms)s should be resended?
+
+## INTRODUCTION
+  [`DMP`](#dmp--delivery-monitoring-protocol) has to be used with the [`TRANSPORT`](#terms) for tracking [`FRAGMENT`](#terms) delivery and, in case of the [`TRANSPORT`](#terms) failure, detecting undelivered [`FRAGMENT`](#terms)s. It solves [`PROBLEM`](#problem), and takes care of the efficient use of memory and the time required for recovery. The protocol is symmetric for the sending and receiving side.
+
+## PROTOCOL DESCRIPTION
+### SPECIAL FRAGMENTS
+- `ACK`: does not contain any information, has to be sent in response to [`ACK`](#special-fragments) from the other side.
+- `SYN`: contains [`REMOTE_STORAGE_PTR`](#data-structures) and [`COUNTER`](#data-structures), and has to be sent primarily after [`TRANSPORT`](#terms) restoration from both sides.
+  
+### DATA STRUCTURES
+- `STORAGE[0]`, `STORAGE[1]`: two ordered "storages" of sent [`FRAGMENT`](#terms)s. (The "storage" is only conditional. The main thing is that you should have access to the already sent [`FRAGMENT`](#terms)s, divided into two parts as the protocol requires.)
+- `LOCAL_STORAGE_PTR`: a boolean pointer to which "storage" the [`FRAGMENT`](#terms)s should be written.
+- `REMOTE_STORAGE_PTR`: a boolean pointer, latest verified local knowledge of the status of a remote [`LOCAL_STORAGE_PTR`](#data-structures).
+- `COUNTER`: an unsigned integer counter of the received data [`FRAGMENT`](#terms)s.
+
+### PROTOCOL STAGES
+#### PREPARATION
+  Before the start of the transferring [`FRAGMENT`](#terms)s both sides have to initialize the necessary data structures with default values (empty storages([`STORAGE[0]`](#data-structures), [`STORAGE[1]`](#data-structures)), [`LOCAL_STORAGE_PTR`](#data-structures)`= 0`, [`REMOTE_STORAGE_PTR`](#data-structures)`= 0`, [`COUNTER`](#data-structures)`= 0`).
+
+#### NORMAL MODE
+  When side sends [`FRAGMENT`](#terms), it has to write a [`FRAGMENT`](#terms) to the [`STORAGE[LOCAL_STORAGE_PTR]`](#data-structures). When receiver side gets [`FRAGMENT`](#terms), it has to increment [`COUNTER`](#data-structures). When anyone gets [`ACK`](#special-fragments) it has to set [`COUNTER`](#data-structures)`= 0` and at the same time negate the [`REMOTE_STORAGE_PTR`](#data-structures), and then send the [`ACK`](#special-fragments) back and at the same time negate the [`LOCAL_STORAGE_PTR`](#data-structures) and clear [`STORAGE[LOCAL_STORAGE_PTR]`](#data-structures).
+
+| Transition | State changes |
+|:-------------:|:----------------:|
+| send [`FRAGMENT`](#terms) | add a [`FRAGMENT`](#terms) to the [`STORAGE[LOCAL_STORAGE_PTR]`](#data-structures) |
```

Comment:
It seems more grammatically correct to write actions in one tense, so that verbs should be in Past Simple (`sent` and `received`) or Present Simple  (`send` and `receive`) at one time.